modular_skyrat/modules/holomap/code/station_holomap.dm

```diff
@@ -0,0 +1,304 @@
+	floor_markings.dir = src.dir
+
+	update_icon()
+
+/obj/machinery/station_map/attack_hand(var/mob/user)
+	if(user == watching_mob)
+		close_map(user)
+		return
+
+	open_map(user)
+
+/// Tries to open the map for the given mob. Returns FALSE if it doesn't meet the criteria, TRUE if the map successfully opened with no runtimes.
+/obj/machinery/station_map/proc/open_map(var/mob/user)
+	if(!anchored || (machine_stat & (NOPOWER | BROKEN)) || !user?.client || panel_open)
+		return FALSE
+
+	if(!holomap_datum)
+		// Something is very wrong if we have to un-fuck ourselves here.
+		message_admins("\[HOLOMAP] WARNING: Holomap at [x], [y], [z] [ADMIN_FLW(src)] had to set itself up on interact! Something during Initialize went very wrong!")
+		setup_holomap()
+
+	holomap_datum.update_map(handle_extra_overlays())
+
+	var/datum/hud/human/user_hud = user.hud_used
+	holomap_datum.base_map.loc = user_hud.holomap  // Put the image on the holomap hud
+	holomap_datum.base_map.alpha = 0 // Set to transparent so we can fade in
+
+	RegisterSignal(user, COMSIG_MOVABLE_MOVED, PROC_REF(check_position))
+
+	playsound(src, 'modular_skyrat/modules/holomap/sound/holomap_open.ogg', 125)
+	animate(holomap_datum.base_map, alpha = 255, time = 5, easing = LINEAR_EASING)
+	icon_state = "station_map_active"
+	set_light(2, 3)
+
+	user.hud_used.holomap.used_station_map = src
+	user.hud_used.holomap.mouse_opacity = MOUSE_OPACITY_ICON
+	user.client.screen |= user.hud_used.holomap
+	user.client.images |= holomap_datum.base_map
+
+	watching_mob = user
+	update_use_power(ACTIVE_POWER_USE)
+
+	if(bogus)
+		to_chat(user, span_warning("The holomap failed to initialize. This area of space cannot be mapped."))
+	else
+		to_chat(user, span_notice("A hologram of the station appears before your eyes."))
+
+	return TRUE
+
+/obj/machinery/station_map/attack_ai(var/mob/living/silicon/robot/user)
+	attack_hand(user)
+
+/obj/machinery/station_map/process()
+	if((machine_stat & (NOPOWER | BROKEN)) || !anchored)
+		close_map()
+
+/obj/machinery/station_map/proc/check_position()
+	SIGNAL_HANDLER
+	if(!watching_mob)
+		return
+
+	if(!Adjacent(watching_mob))
+		close_map(watching_mob)
+
+/obj/machinery/station_map/proc/close_map()
+	if(!watching_mob)
+		return
+
+	UnregisterSignal(watching_mob, COMSIG_MOVABLE_MOVED)
+	playsound(src, 'modular_skyrat/modules/holomap/sound/holomap_close.ogg', 125)
+	icon_state = initial(icon_state)
+	if(watching_mob.client)
+		animate(holomap_datum.base_map, alpha = 0, time = 5, easing = LINEAR_EASING)
+		spawn(5) //we give it time to fade out
+			watching_mob.client.screen -= watching_mob.hud_used.holomap
+			watching_mob.client.images -= holomap_datum.base_map
+			watching_mob.hud_used.holomap.used_station_map = null
+			watching_mob = null
+			set_light(1, 2)
+
+	update_use_power(IDLE_POWER_USE)
+
+/obj/machinery/station_map/power_change()
+	. = ..()
+	update_icon()
+
+	if(machine_stat & NOPOWER)
+		set_light(0)
+	else
+		set_light(1, 2)
```

Comment:
i feel like this could end up with inconsistency in the future from values being magical

Reply:
See title.